Add concurrency to e2e-upgrade tests

aanm opened the pull request:
The changes on this PR helps to decrease the test run time significantly. During testing if this was possible, a lot of flakes were being caused by the egress-gateway, egress-gateway-with-l7-policy and from-cidr-host-netns tests which is why they run separately without concurrency and then all the other tests run parallel afterwards.

You can find 6 successful runs in https://github.com/cilium/cilium/actions/runs/11220270835. The attempt #4 that failed was caused by a faulty connection to one.one.one.one.

| Name | Before    | After     | Time Decreased |
|-----------|-----------|-----------|----------------|
| Test 1    | 00:49:18  | 00:25:14  | -48.82%        |
| Test 2    | 00:46:53  | 00:24:37  | -47.49%        |
| Test 3    | 00:47:12  | 00:25:14  | -46.54%        |
| Test 4    | 00:55:41  | 00:32:41  | -41.30%        |
| Test 5    | 00:52:23  | 00:28:58  | -44.70%        |
| Test 6    | 01:07:00  | 00:38:44  | -42.19%        |
| Test 7    | 01:10:00  | 00:39:11  | -44.02%        |
| Test 8    | 00:56:24  | 00:32:19  | -42.70%        |
| Test 9    | 00:54:19  | 00:32:27  | -40.26%        |
| Test 10   | 00:48:46  | 00:25:52  | -46.96%        |
| Test 11   | 01:04:00  | 00:31:34  | -50.68%        |
| Test 12   | 01:03:00  | 00:37:17  | -40.82%        |
| Test 13   | 00:46:53  | 00:22:37  | -51.76%        |
| Test 14   | 01:01:00  | 00:37:01  | -39.32%        |
| Test 15   | 00:58:01  | 00:32:17  | -44.36%        |
| Test 16   | 01:02:00  | 00:35:16  | -43.12%        |
| Test 17   | 01:01:00  | 00:33:11  | -45.60%        |
| Test 18   | 00:59:33  | 00:31:09  | -47.69%        |
| Test 19   | 01:02:00  | 00:34:44  | -43.98%        |
| Test 20   | 01:04:00  | 00:35:07  | -45.13%        |
| Test 21   | 01:01:00  | 00:39:52  | -34.64%        |
| Test 22   | 01:05:00  | 00:35:15  | -45.77%        |
| Test 23   | 00:30:29  | 00:16:50  | -44.78%        |

aanm committed:
.github: add concurrency to tests-e2e-upgrade workflow

Adding concurrency to tests-e2e upgrade workflow helps to decrease the
time it takes to run the tests on our CI.

| Name      | Before    | After     | Time Decreased |
|-----------|-----------|-----------|----------------|
| Test 1    | 00:49:18  | 00:25:14  | -48.82%        |
| Test 2    | 00:46:53  | 00:24:37  | -47.49%        |
| Test 3    | 00:47:12  | 00:25:14  | -46.54%        |
| Test 4    | 00:55:41  | 00:32:41  | -41.30%        |
| Test 5    | 00:52:23  | 00:28:58  | -44.70%        |
| Test 6    | 01:07:00  | 00:38:44  | -42.19%        |
| Test 7    | 01:10:00  | 00:39:11  | -44.02%        |
| Test 8    | 00:56:24  | 00:32:19  | -42.70%        |
| Test 9    | 00:54:19  | 00:32:27  | -40.26%        |
| Test 10   | 00:48:46  | 00:25:52  | -46.96%        |
| Test 11   | 01:04:00  | 00:31:34  | -50.68%        |
| Test 12   | 01:03:00  | 00:37:17  | -40.82%        |
| Test 13   | 00:46:53  | 00:22:37  | -51.76%        |
| Test 14   | 01:01:00  | 00:37:01  | -39.32%        |
| Test 15   | 00:58:01  | 00:32:17  | -44.36%        |
| Test 16   | 01:02:00  | 00:35:16  | -43.12%        |
| Test 17   | 01:01:00  | 00:33:11  | -45.60%        |
| Test 18   | 00:59:33  | 00:31:09  | -47.69%        |
| Test 19   | 01:02:00  | 00:34:44  | -43.98%        |
| Test 20   | 01:04:00  | 00:35:07  | -45.13%        |
| Test 21   | 01:01:00  | 00:39:52  | -34.64%        |
| Test 22   | 01:05:00  | 00:35:15  | -45.77%        |
| Test 23   | 00:30:29  | 00:16:50  | -44.78%        |

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: tests-e2e-upgrade run flaky tests sequentially

Since a lot of flakes were being caused by these tests we need to run
them separately first and then run all the other tests concurrently.

We will rename the tests that need to be run sequentially by prefixing
their names with "seq-".

Signed-off-by: André Martins <andre@cilium.io>

michi-covalent reviewed (approved):
very concurrent

brb commented on .github/workflows/tests-e2e-upgrade.yaml:
```diff
@@ -585,6 +588,24 @@ jobs:
             --include-conn-disrupt-test \
             --conn-disrupt-test-restarts-path "./cilium-conn-disrupt-restarts" \
             --flush-ct \
+            --test "seq-.*" \
```
~I haven't been recently following the CLI development. What is this `seq-.*`?~ OK, just saw the changes at the bottom.

brb replied:
`--test "seq-.*" ` - won't this prevent from testing `no-interrupted-connections`?

aanm replied:
@brb Thanks for the review. After the last changes I'm seeing
```
[=] [cilium-test-1] Test [no-interrupted-connections] [1/105]
```

for the lines that are exclusively running the `--test "no-interrupted-connections"`.

I assume this is what we want?

aanm committed:
.github: re-structure e2e-upgrade for clarity

Splitting the connectivity tests commands together with comments will
make it clear what are we testing as this was causing some confusion
which tests were being tested on each "cilium connectivity" invocation.

Signed-off-by: André Martins <andre@cilium.io>

julianwiedmann reviewed (approved):
nice win! And running the tests in parallel feels like we're one step closer to real-world chaos testing :).

The sequential tests feel a bit ad-hoc, would be good to understand better why these tests need special treatment. Could point us at overall instability with some feature combinations ...

brb commented on cilium-cli/connectivity/builder/from_cidr_host_netns.go:
```diff
@@ -12,7 +12,7 @@ import (
 
 func (t fromCidrHostNetns) build(ct *check.ConnectivityTest, templates map[string]string) {
-	newTest("from-cidr-host-netns", ct).
+	newTest("seq-from-cidr-host-netns", ct).
```
Could you add a code comment saying why we have the `seq-` prefix?

brb reviewed (approved):
Thanks! Some non-blocking nit.

brlbil reviewed (approved):
timeout-minutes: 90 -> 45 🤩

aanm committed:
cilium-cli: increase wait timeout for services

With the introduction of concurrent tests, it takes a little bit more
time for all the services to be propagated. Thus we need to increase the
timeout from 30 seconds to 1 minute.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
cilium-cli: increase wait timeout for policy enforcement

With the introduction of concurrent tests, it takes a little bit more
time for all the policies to be propagated. Thus we need to increase the
timeout from 15 seconds to 30 seconds.

Signed-off-by: André Martins <andre@cilium.io>
